src/pypromice/pipeline/L1toL2.py

```diff
@@ -140,20 +140,24 @@ def toL2(
     else:
         precip_flag=True
     if ~ds["precip_u"].isnull().all() and precip_flag:
-        ds["precip_u"] = precipitation.filter(ds["precip_u"], ds["t_u"], ds["p_u"], ds["rh_u"])
-        ds["precip_rate_u"] = precipitation.convert_to_rate(ds["precip_u"], ds["wspd_u"], ds["t_u"])
+        ds["precip_u"] = precipitation.filter_lufft_errors(ds["precip_u"], ds["t_u"], ds["p_u"], ds["rh_u"])
+        ds["rainfall_u"], ds["rainfall_cor_u"] = \
```

Comment:
See comment about splitting up `precipitation.convert_to_rainfall_per_timestep_and_correct_undercatch`

```suggestion
        ds["rainfall_u"] = get_rainfall_per_timestep(precip, ds["t_u"])
        ds["rainfall_cor_u"] = correct_rainfall_undercatch(ds["rainfall_u"], ds["wpsd_u"])
```